lib/Define.ts

```diff
@@ -0,0 +1,65 @@
+import {
+    Message,
+    MessageEmbed,
+} from 'discord.js';
+
+import fetch from 'node-fetch';
+
+import {
+    Paginate,
+    DisplayType,
+} from './Paginate.js';
+
+
+export async function handleDefine(msg: Message, args: string): Promise<void> {
+    let query = args.trim();
+
+    /* Prevent leaking our IP */
+    if (query.includes('auto:ip')) {
+        query = '';
+    }
+
+    if (query === '') {
+        msg.reply(`No word given.`);
+        return;
+    }
+
+    query = query.replaceAll(" ", "%20");
+    
+    const url = `https://api.dictionaryapi.dev/api/v2/entries/en/${query}`;
+
+    try {
+
+        const res = await fetch(url);
+        const data = await res.json();
+
+        const embed = new MessageEmbed();
+        const meanings = data[0].meanings
+
+        
+        let definitions = [];
+        for (const meaning of meanings) {
+            for (const partOfSpeech of meaning.definitions) {
+                definitions.push({
+                    name: meaning.partOfSpeech,
+                    value: partOfSpeech.definition,
+                    inline: true,
+                });
+            }
+        }
+
+        const pages = new Paginate({
+            sourceMessage: msg,
+            itemsPerPage: 9,
+            displayFunction: (item: any) => item,
+            displayType: DisplayType.EmbedFieldData,
+            data: definitions,
+            embed,
+        });
+    
+
+        await pages.sendMessage();
+    } catch (err) {
+        msg.reply(`Word or phrase is not in the dictionary`);
```

Comment:
Need to await any discord API functions that return a promise otherwise error messages won't be correctly caught
```suggestion
        await msg.reply(`Word or phrase is not in the dictionary`);
```